pkg/virt-controller/watch/vmi.go

```diff
@@ -2100,3 +2111,36 @@ func (c *VMIController) getVolumePhaseMessageReason(volume *virtv1.Volume, names
 	}
 	return virtv1.VolumePending, PVCNotReadyReason, "PVC is in phase Lost"
 }
+
+func (c *VMIController) handleDynamicInterfaceRequests(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	log.Log.Object(pod).V(4).Info("network iface hotplug for pod")
+	multusAnnotation, err := services.GenerateMultusCNIAnnotation(vmi)
+	if err != nil {
+		return err
+	}
+
+	currentMultusAnnotation, found := pod.Annotations[networkv1.NetworkAttachmentAnnot]
+	if found {
+		log.Log.Object(pod).V(4).Infof("current multus annotation for pod: %s", currentMultusAnnotation)
+	}
+	if multusAnnotation != "" {
+		pod.Annotations[networkv1.NetworkAttachmentAnnot] = multusAnnotation
+		log.Log.Object(pod).V(4).Infof("updated multus annotation for pod: %s", multusAnnotation)
+	} else {
+		delete(pod.Annotations, networkv1.NetworkAttachmentAnnot)
+		log.Log.Object(pod).V(4).Info("cleared pod cncf.io/networks annotation")
+	}
+
+	if currentMultusAnnotation != multusAnnotation {
+		pod, err = c.clientset.CoreV1().Pods(pod.GetNamespace()).UpdateStatus(context.Background(), pod, v1.UpdateOptions{})
```

Comment:
This seems risky to me with a potential to collide.
Why it is not patched?

A pod status is frequently updated all over, by many controllers.

Reply:
Well, I do not even understand how `UpdateStatus` has the permissions to touch an annotation (which is metadata). I thought `UpdateStatus` can only touch the `Status` subtree of the manifest.

Reply:
I am no longer using `updateStatus` - I use `Update` instead, which enables me to rely on the currently configured RBAC rules for the virt-controller.

I'll try to change it to a `patch` later on.

Reply:
Done, I am now re-using `syncPodAnnotations`, which under the hood patches the pod.